go/types/list.go

```diff
@@ -116,15 +135,21 @@ func (l List) elemType() *Type {
 	return l.seq.Type().Desc.(CompoundDesc).ElemTypes[0]
 }
 
+// Set returns a new list where the valie at the given index have been replaced with v. If idx is
+// out bounds then this panics.
 func (l List) Set(idx uint64, v Value) List {
 	d.Chk.True(idx < l.Len())
 	return l.Splice(idx, 1, v)
 }
 
+// Append returns a new list where vs have bben appended to the resulting list.
 func (l List) Append(vs ...Value) List {
 	return l.Splice(l.Len(), 0, vs...)
 }
 
+// Splice returns a new list where deleteCount values have been removed at idx and vs have been
+// inserted instead.
+// This function panics if idx is out of bounds or too many items are tried to be removed.
```

Comment:
You could also say "This function panics if idx or idx + deleteCount is out of bounds."